src/Ticket.php

```diff
@@ -6220,10 +6232,15 @@ public function showStatsDates()
     {
         $now                      = time();
         $date_creation            = strtotime($this->fields['date'] ?? '');
+       // Tickets created before 10.0.4 do not have takeintoaccountdate field, use old and incorrect computation for those cases
         $date_takeintoaccount     = $date_creation + $this->fields['takeintoaccount_delay_stat'];
         if ($date_takeintoaccount == $date_creation) {
             $date_takeintoaccount  = 0;
         }
+       // If takeintoaccountdate was saved use it
+        if(!empty($this->fields['takeintoaccountdate'])) {
+            $date_takeintoaccount = strtotime($this->fields['takeintoaccountdate']);
+        }
```

Comment:
It would be easier to read, IMHO.
```suggestion
        $date_takeintoaccount     = 0;
        if ($this->fields['takeintoaccountdate'] !== null) {
            $date_takeintoaccount = strtotime($this->fields['takeintoaccountdate']);
        } elseif ($this->fields['takeintoaccount_delay_stat'] > 0) {
            $date_takeintoaccount = $date_creation + $this->fields['takeintoaccount_delay_stat'];
        }
```